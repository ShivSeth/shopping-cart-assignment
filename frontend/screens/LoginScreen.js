import React, { useState } from 'react';
import { Redirect, useHistory } from 'react-router';
import { useAuthContext } from '../context/AuthContext';
import './LoginScreen.scss';

const LoginScreen = () => {
  const [input, setInput] = useState({ email: '', password: '' });
  const { authState, setAuthState, checkAuthState } = useAuthContext();

  const [errorMsg, setErrorMsg] = useState('');
  const history = useHistory();
  function formSubmit(e) {
    e.preventDefault();
    if (input.email !== '' && input.password !== '') {
      const status = checkAuthState(input.email, input.password);
      if (status) {
        setAuthState(status);
        history.push('/');
      } else {
        setErrorMsg('Inputs incorrect');
        setTimeout(() => {
          setErrorMsg('');
        }, 3000);
      }
    } else {
      setErrorMsg('Inputs incorrect');
      setTimeout(() => {
        setErrorMsg('');
      }, 3000);
    }
  }

  return (
    <div className="login-form">
      <aside className="left-side">
        <h2>Sign In</h2>
        <p>Get access to your orders, wishlist and recommendations</p>
      </aside>
      <form className="right-side" onSubmit={formSubmit}>
        <input
          required
          type="email"
          name="email"
          placeholder="Email"
          onChange={(e) => setInput({ ...input, email: e.target.value })}
        />
        <input
          pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}"
          minLength="6"
          required
          type="password"
          name="password"
          placeholder="Password"
          onChange={(e) => setInput({ ...input, password: e.target.value })}
        />
        {errorMsg && <p className="error-msg">{errorMsg}</p>}

        <button type="submit">Sign In</button>
      </form>
    </div>
  );
};

export default LoginScreen;
