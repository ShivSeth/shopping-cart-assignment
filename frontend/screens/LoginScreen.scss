.login-form {
  margin-top: 10rem;
  display: flex;
  justify-content: center;

  input {
    display: block;
  }

  .left-side {
    padding: 0 1.5rem;
    margin: 10px;
    h2 {
      margin-bottom: 1.5rem;
    }
  }

  .right-side {
    padding: 0 1.5rem;
    width: 30%;

    .error-msg {
      text-align: center;
      color: red;
      margin-bottom: 1rem;
    }

    input {
      width: 100%;
      padding: 10px 0;
      margin-bottom: 1rem;
      border: none;
      outline: none;
      border-bottom: 1px solid gray;

      &:focus {
        border-bottom: 2px solid #51dae2;
      }
    }

    button {
      width: 100%;
      border: none;
      background-color: #d00156;
      padding: 10px;
      color: azure;

      &:hover {
        cursor: pointer;
      }
    }
  }
}

@media screen and (max-width: 600px) {
  .login-form {
    flex-flow: column;

    justify-content: center;
    align-items: center;

    .left-side,
    .right-side {
      width: 80%;
    }

    .right-side {
      margin-top: 1rem;
    }
  }
}
