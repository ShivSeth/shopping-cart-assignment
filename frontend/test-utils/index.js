import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { AuthContextProvider } from '../context/AuthContext';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import store from '../store';
store;
const AllTheProviders = ({ children }) => {
  const history = createMemoryHistory();

  return (
    <Provider store={store}>
      <AuthContextProvider>
        <Router history={history}>{children}</Router>
      </AuthContextProvider>
    </Provider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
