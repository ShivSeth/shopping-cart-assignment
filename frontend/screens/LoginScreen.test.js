import { render, screen } from '../test-utils';
import LoginScreen from './LoginScreen';

describe('<LoginScreen/>', () => {
  test('should sign in input boxes be present ', () => {
    const { container } = render(<LoginScreen />);
    const emailInput = screen.getByPlaceholderText('Email');
    const passwordInput = screen.getByPlaceholderText('Password');
    expect(emailInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
