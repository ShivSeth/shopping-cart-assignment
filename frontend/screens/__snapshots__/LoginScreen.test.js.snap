// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen/> should sign in input boxes be present  1`] = `
<div>
  <div
    class="login-form"
  >
    <aside
      class="left-side"
    >
      <h2>
        Sign In
      </h2>
      <p>
        Get access to your orders, wishlist and recommendations
      </p>
    </aside>
    <form
      class="right-side"
    >
      <input
        name="email"
        placeholder="Email"
        required=""
        type="email"
      />
      <input
        minlength="6"
        name="password"
        pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z]).{6,}"
        placeholder="Password"
        required=""
        type="password"
      />
      
      <button
        type="submit"
      >
        Sign In
      </button>
    </form>
  </div>
</div>
`;
